src/frontend/src/views/Upload/components/Metadata/components/StaticTable/index.tsx

```diff
@@ -50,78 +62,132 @@ const validationSchema = object({
 export default function StaticTable({
   metadata,
   setIsValid,
+  hasImportedMetadataFile,
 }: Props): JSX.Element {
+  const [validationErrors, setValidationErrors] =
+    useState<Record<string, ValidationErrorRecord | null>>(EMPTY_OBJECT);
+
+  // This function validates metadata by creating a mapping of sample ids to
+  // either null or a ValidationErrorRecord, an object which is itself a mapping
+  // from the key(s) that failed to validate to the specific error message.
+  // The function then determines if the metadata as a whole is valid by
+  // checking that each sampleId maps to null.
+  // If one field has multiple error messages we just overwrite the message in
+  // the ValidationErrorRecord, since we only want to display one error at a time.
   const validateMetadata = useCallback(
     async (metadata: SampleIdToMetadata | null) => {
       if (metadata == null) {
         setIsValid(false);
         return;
       }
-      const rowValidation: Record<string, boolean> = {};
+      const validationErrors: Record<string, ValidationErrorRecord | null> =
+        Object.fromEntries(
+          Object.keys(metadata).map((sampleId) => [sampleId, null])
+        );
       for (const [sampleId, sampleMetadata] of Object.entries(metadata)) {
-        const isRowValid = await validationSchema.isValid(sampleMetadata);
-        rowValidation[sampleId] = isRowValid;
+        try {
+          await validationSchema.validate(sampleMetadata, {
+            abortEarly: false,
+          });
+        } catch (error) {
+          if (error instanceof ValidationError) {
+            const errorRecord: ValidationErrorRecord = {};
+            error.inner.forEach((validationError) => {
+              if (validationError.path != undefined) {
+                const rootMetadataKey = validationError.path.split(".")[0];
+                errorRecord[rootMetadataKey as keyof ValidationErrorRecord] =
```

Comment:
is this type assertion required?

Reply:
Yeah, I only put the type assertions in when TS was making noises at me.

Reply:
Actually, on a second thought I just changed the type definition to make this assertion go away.

Reply:
Yea, typically that is the way to go. A type assertion shouldn't be used as a panacea for type compiler complaints. Usually when the type checker fails it's because you have two types somewhere that don't line up properly.